### ISSUE TITLE
How do language codes work?

### ISSUE THREAD

**Issue by victorgveloso** [arvida42/jackettio#34]
In Jackett Description files (https://github.com/Jackett/Jackett/tree/master/src/Jackett.Common/Definitions). The field `language` follows the format language-COUNTRY (e.g., en-US, es-ES, fr-FR, pt-BR).  However, in https://github.com/arvida42/jackettio/blob/master/src/lib/config.js#L88-L103 you specify `iso639` as a two letter field (e.g., en, es, fr, pt) and `pattern` as a regex pattern.

For instance, when I prioritize `pt` (portuguese) in Jackettio, I barely get any results. Even though my self-hosted Jackett server includes brazilian indexers, which contain the contents I am searching for (e.g., Fullmetal Alchemist Brotherhood is available in uniotaku.com). I wonder if it's a limitation of the pt pattern `portuguese`. To suggest some improvements, I collected some patterns I have found being used to refer to Brazilian Portuguese in the wild:

- `pob` (used by stremio)
- `pt-BR` (used by jackett)
- `ptbr`
- `br` (common short name in embedded mkv subtitles)
- `brazilian`

Could you confirm that the portuguese priority language setting comprises all of those patterns?

**Comment by arvida42** [comment]
The language is determined based on the torrent title. I chose not to use the filter in Jackett directly, as in my experience, it's not always accurate and doesn't consistently return the desired results. Even including the year parameter in a Jackett search can return strange results.

To improve language prioritization in search, I need the patterns used in titles for your language.

**Comment by victorgveloso** [comment]
Brazilian Portuguese will look like Spanish in torrent titles so you can expect `Dublado`, `Dual Áudio`, `Dual Audio` as indicators for dubbed streams. Besides that, brazilian portuguese is a supported language in TMDB. 

Actually, the TMDB addon author is a Brazilian himself so the main supported language is Brazilian Portuguese. I would expect that addon to support it very well.

**Comment by victorgveloso** [comment]
Hi @arvida42,

As shown below, Jackettio queries in english and I get more results when I manually type the search query in portuguese ("exterminador do futuro") than when I use your add-on.

```
jackett  | 11-19 11:27:52 Info Torznab search in UniOtaku for The Terminator => Found 0 releases [183ms]
jackett  | 11-19 11:27:52 Info Torznab search in LAPUMiA for The Terminator => Found 0 releases [493ms]
jackett  | 11-19 11:27:53 Info Torznab search in ApacheTorrent for The Terminator => Found 0 releases [1309ms]
jackett  | 11-19 11:27:53 Info Torznab search in RedeTorrent for The Terminator => Found 0 releases [1473ms]
jackett  | 11-19 11:27:54 Info Torznab search in Filmes HD Torrent for The Terminator => Found 0 releases [2390ms]
jackett  | 11-19 11:28:00 Info Torznab search in BluDV for The Terminator => Found 3 releases [8537ms]
jackett  | 11-19 11:28:00 Info Torznab search in TorrentDosFilmes for The Terminator => Found 3 releases [8677ms]
jackett  | 11-19 11:29:09 Info Manual search in ApacheTorrent, BluDV, Filmes HD Torrent, LAPUMiA, RedeTorrent, TorrentDosFilmes, UniOtaku for exterminador do futuro => Found 26 releases [14512ms]
```

Both my docker-compose and the jackettio/configure are set to portuguese. The former, through the env var DEFAULT_META_LANGUAGE=pt, the latter through the selection box as shown in the screenshot below:

<img width="582" alt="image" src="https://github.com/user-attachments/assets/95085fb9-7f4c-4712-809d-379acb50cc12">

This is an important issue to me who just developed six indexers for Portuguese dubbed content support in Jackett. If you need any more information, please let me know.

**Comment by victorgveloso** [comment]
Ok, I figured it out. DEFAULT_META_LANGUAGE env var is overridden by the cofigure webpage selection widget. 

OTOH, I found out that TMDB actually support brazilian portuguese. Actually, in their documentation they use brazilian portuguese as the example to demonstrate that.

All you need to do is add the country suffix (https://developer.themoviedb.org/docs/languages#iso-3166-1) to the language code (https://developer.themoviedb.org/docs/languages#iso-3166-1) so, for example, pt-br corresponds to brazilian portuguese and pt stands for Portuguese from Portugal. The same applies to other languages but probably portuguese has the most movie title variation among countries.

You can try yourself in [their api swagger.io](https://developer.themoviedb.org/reference/find-by-id). Type "pt-br" for language and "tt6587046" for imdb_id. Then, change to "pt". The title changes from "O Menino e a Garça" to "O Rapaze a Garça"

The reason I was getting fewer results is due to that difference between movie titles between Portugal and Brazil.

In summary, all you need to do is accept the PR #37

**Comment by arvida42** [comment]
You have to configure the `TMDB_ACCESS_TOKEN` to have access to "search language"
